app/models/user.rb

```diff
@@ -1,6 +1,21 @@
 class User < ApplicationRecord
-  has_many :votes
+  has_many :votes, dependent: :destroy
   has_many :ranked_works, through: :votes, source: :work
 
   validates :username, uniqueness: true, presence: true
+
+  def self.build_from_github(auth_hash)
+
+    user_data = {
+      uid: auth_hash[:uid],
+      username: auth_hash[:info][:username],
+      email: auth_hash[:info][:email],
+      provider: auth_hash[:provider]
+    }
+
+    user = self.new(user_data)
+    if user.save
+      return user
```

Comment:
You call `user.save` both here and in your controller. You should pick one or the other. Based on the structure of your code, I think the controller would be a better choice - that way you can give information to the user if something goes wrong.